plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/machine/MachineStatusHandler.java

```diff
@@ -119,6 +120,12 @@ private void handleMachineRunning(final String machineId, final WorkspaceRuntime
         if (machine == null) {
             return;
         }
-        eventBus.fireEvent(new MachineStateEvent(machine, RUNNING));
+
+        try {
+            eventBus.fireEvent(new MachineStateEvent(machine, RUNNING));
```

Comment:
Yes. To have a special event bus which guarantee each registered handler will be invoked would be a supergreat!